src/armory/Armory.js

```diff
@@ -64,159 +68,188 @@ class Armory extends React.Component<Props, State> {
+			
 			userId: '',
+			tankOptions : [new OptionClass('','')],
+			chassisOptions : [new OptionClass('','')],
+			weaponOneOptions : [new OptionClass('','')],
+			weaponTwoOptions: [new OptionClass('','')],
+			scannerOneOptions: [new OptionClass('','')],
+			scannerTwoOptions: [new OptionClass('','')],
+			scannerThreeOptions: [new OptionClass('','')],
+			jammerOptions: [new OptionClass('','')],
+			treadsOptions: [new OptionClass('','')],
+			singleUseItemsOne: [new OptionClass('','')],
+			singleUseItemsTwo: [new OptionClass('','')],
+			singleUseItemsThree: [new OptionClass('','')],
 		}
 		this.getFavoriteTank();
 	}
 	//Clears the inventory on the frontend side
 	clearInventoryArrays() {
-		tankOptions = [{value: '', label: ''}];
-		chassisOptions = [{value: '', label: ''}];
-		weaponOneOptions = [{value: '', label: ''}];
-		weaponTwoOptions = [{value: '', label: ''}];
-		scannerOneOptions= [{value: '', label: ''}];
-		scannerTwoOptions= [{value: '', label: ''}];
-		scannerThreeOptions= [{value: '', label: ''}];
-		jammerOptions = [{value: '', label: ''}];
-		treadsOptions = [{value: '', label: ''}];
-		singleUseItemsOne = [{value: '', label: ''}];
-		singleUseItemsTwo = [{value: '', label: ''}];
-		singleUseItemsThree = [{value: '', label: ''}];
+		this.setState({
+			tankOptions : [new OptionClass('','')],
+			chassisOptions : [new OptionClass('','')],
+			weaponOneOptions : [new OptionClass('','')],
+			weaponTwoOptions: [new OptionClass('','')],
+			scannerOneOptions: [new OptionClass('','')],
+			scannerTwoOptions: [new OptionClass('','')],
+			scannerThreeOptions: [new OptionClass('','')],
+			jammerOptions: [new OptionClass('','')],
+			treadsOptions: [new OptionClass('','')],
+			singleUseItemsOne: [new OptionClass('','')],
+			singleUseItemsTwo: [new OptionClass('','')],
+			singleUseItemsThree: [new OptionClass('','')],
+		});
 	}
+
 	//This is used to get the current favorite tank of the user and continues to get all of the selected tank
-	getFavoriteTank = async ():Promise<void> => {
-		const cookies = new Cookies();
-		const token = cookies.get('token');
-		const response = await fetch('/api/tank/getFavorite/', {
-			method: 'GET',
-			headers: {
-				'Access-Control-Allow-Origin': '*',
-				'Content-Type': 'application/json',
-				'Access-Control-Allow-Credentials': 'true',
-				'x-auth-token': token
-			},
-		});
-		const body = await response.text();
-		this.setState({selectedTankId: body});
-		this.getSelectedTank();
+	getFavoriteTank () : void {
+		const responsePromise = getFavoriteTankID();
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+				}
+				else {
+					const favoriteTankID = data.text();
+					this.setState({selectedTankId: favoriteTankID});
+					this.getSelectedTank();
+				}
+			})
+		).catch(
+			error => {
+				console.log('No favorite Tank');
```

Comment:
Is this true? If the response fails it is likely due to a connection error, not something being wrong on the server, right?

Reply:
Just to have an answer on this, it happens because if the user doesn't have a favorite tank the api returns nothing so the responsePromise.then gives a end of json object error.